packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1347,15 +1446,26 @@ class FollowerLayer extends ContainerLayer {
     _inverseDirty = true;
   }
 
+  /// {@template flutter.leaderFollower.markClean}
+  /// We intentionally make this method do nothing so LeaderLayer and
+  /// FollowerLayer will always be in dirty state and we won't apply retained
+  /// rendering to those layers. We have to do this because they break the
+  /// assumption of retained rendering: even if no layer in a subtree is dirty,
+  /// the subtree may still be dirty because the FollowerLayer copies change
+  /// from a LeaderLayer that could be anywhere in the Layer tree.
+  /// {@endtemplate}
+  @override
+  void markClean() {}
```

Comment:
this seems like a weird way to have that effect. Maybe we should just have a getter that returns true or false based on whether a node supports being retained?